factory/alchemy.py

```diff
@@ -22,11 +22,17 @@ def _check_sqlalchemy_session_persistence(self, meta, value):
                 (meta, VALID_SESSION_PERSISTENCE_TYPES, value)
             )
 
+    def _check_has_sqlalchemy_session_set(self, meta, value):
+        if value and getattr(meta, "sqlalchemy_session"):
```

Comment:
```suggestion
        if value and meta.sqlalchemy_session:
```

Reply:
Done in [a89c62c](https://github.com/FactoryBoy/factory_boy/pull/927/commits/a89c62c94b6172129c2d85af96acd9a98f477bfb)